### PR TITLE
reposurgeon: 3.44 -> 4.26

### PR DESCRIPTION
**Pull request by dguibert**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 122854` at bd752b94 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>reposurgeon</li>
  </ul>
</details>
<details>
  <summary>11 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `monotone`.
Near pkgs/applications/version-management/reposurgeon/default.nix:4:21:
```
  |
4 | , mercurial ? null, monotone ? null, rcs ? null
  |                     ^
```
</li>
    <li>warning: unused-argument

Unused argument: `cvs_fast_export`.
Near pkgs/applications/version-management/reposurgeon/default.nix:5:22:
```
  |
5 | , subversion ? null, cvs_fast_export ? null }:
  |                      ^
```
</li>
    <li>warning: unused-argument

Unused argument: `makeWrapper`.
Near pkgs/applications/version-management/reposurgeon/default.nix:1:26:
```
  |
1 | { lib, stdenv, fetchurl, makeWrapper, buildGoModule, git
  |                          ^
```
</li>
    <li>warning: unused-argument

Unused argument: `cvs`.
Near pkgs/applications/version-management/reposurgeon/default.nix:3:18:
```
  |
3 | , breezy ? null, cvs ? null, darcs ? null, fossil ? null
  |                  ^
```
</li>
    <li>warning: unused-argument

Unused argument: `subversion`.
Near pkgs/applications/version-management/reposurgeon/default.nix:5:3:
```
  |
5 | , subversion ? null, cvs_fast_export ? null }:
  |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `mercurial`.
Near pkgs/applications/version-management/reposurgeon/default.nix:4:3:
```
  |
4 | , mercurial ? null, monotone ? null, rcs ? null
  |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `darcs`.
Near pkgs/applications/version-management/reposurgeon/default.nix:3:30:
```
  |
3 | , breezy ? null, cvs ? null, darcs ? null, fossil ? null
  |                              ^
```
</li>
    <li>warning: unused-argument

Unused argument: `rcs`.
Near pkgs/applications/version-management/reposurgeon/default.nix:4:38:
```
  |
4 | , mercurial ? null, monotone ? null, rcs ? null
  |                                      ^
```
</li>
    <li>warning: unused-argument

Unused argument: `breezy`.
Near pkgs/applications/version-management/reposurgeon/default.nix:3:3:
```
  |
3 | , breezy ? null, cvs ? null, darcs ? null, fossil ? null
  |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `git`.
Near pkgs/applications/version-management/reposurgeon/default.nix:1:54:
```
  |
1 | { lib, stdenv, fetchurl, makeWrapper, buildGoModule, git
  |                                                      ^
```
</li>
    <li>warning: unused-argument

Unused argument: `fossil`.
Near pkgs/applications/version-management/reposurgeon/default.nix:3:44:
```
  |
3 | , breezy ? null, cvs ? null, darcs ? null, fossil ? null
  |                                            ^
```
</li>
  </ul>
</details>

File: pkgs/applications/version-management/reposurgeon/default.nix [hunk 1]
```diff
@@ -1,60 +1,39 @@
-{ lib, stdenv, fetchurl, makeWrapper, python27Packages, git
-, docbook_xml_dtd_412, docbook_xsl, asciidoc, xmlto, pypy
+{ lib, stdenv, fetchurl, makeWrapper, buildGoModule, git
+, asciidoctor, ruby
 , breezy ? null, cvs ? null, darcs ? null, fossil ? null
 , mercurial ? null, monotone ? null, rcs ? null
 , subversion ? null, cvs_fast_export ? null }:
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/reposurgeon/default.nix:5]:
```suggestion
, breezy, cvs, darcs, fossil, mercurial, monotone, rcs, subversion, cvs_fast_export
}:
```

Comment (dguibert) [reply pkgs/applications/version-management/reposurgeon/default.nix:5]: These dependencies are not used

File: pkgs/applications/version-management/reposurgeon/default.nix [hunk 2]
```diff
@@ -1,60 +1,40 @@
...
+
+  meta = {
+    description = "A tool for editing version-control repository history";
+    license = licenses.bsd3;
+    homepage = "http://www.catb.org/esr/reposurgeon/";
+    maintainers = with maintainers; [ dfoxfranke ];
+    platforms = platforms.all;
+  };
+
 }
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/reposurgeon/default.nix:40]:
```suggestion
  };
}
```

File: pkgs/applications/version-management/reposurgeon/default.nix [hunk 3]
```diff
@@ -1,60 +1,40 @@
-{ lib, stdenv, fetchurl, makeWrapper, python27Packages, git
-, docbook_xml_dtd_412, docbook_xsl, asciidoc, xmlto, pypy
-, breezy ? null, cvs ? null, darcs ? null, fossil ? null
-, mercurial ? null, monotone ? null, rcs ? null
-, subversion ? null, cvs_fast_export ? null }:
+{ lib, stdenv, fetchurl, makeWrapper, buildGoModule, git
+, asciidoctor, ruby
+}:
 
 with stdenv; with lib;
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/reposurgeon/default.nix:5]:
```suggestion
```

please add them where they are necessary.

Comment (dguibert) [reply pkgs/applications/version-management/reposurgeon/default.nix]: applied

**Pull request merged**